src/argowrapper/engine/argo_engine.py

```diff
@@ -405,7 +418,7 @@ def get_workflows_for_team_project(self, team_project: str) -> List[Dict]:
         workflows = self.get_workflows_for_label_selector(label_selector=label_selector)
         return workflows
 
-    def get_workflows_for_user(self, auth_header: str) -> List[Dict]:
+    def get_workflows_for_user(self, auth_header: Optional[str]) -> List[Dict]:
```

Comment:
```suggestion
    def get_workflows_for_user(self, auth_header: str) -> List[Dict]:
```
I don't think `auth_header` is optional here.

Reply:
It's because of this, the `get` can return `None`, so it expects it to be `Optional`:
```
auth_header=request.headers.get("Authorization"),
```

In short, we need better type annotations.